addons/account/static/src/js/tax_group.js

```diff
@@ -75,97 +144,93 @@ odoo.define('account.tax_group', function (require) {
-            var $input = $taxGroupElement.find('.tax_group_edit_input input');
-            // Get original value and display it in user locale in the input
-            var formatedOriginalValue = fieldUtils.format.float($input.data('originalValue'), {}, {});
-            $input.focus(); // Focus the input
-            $input.val(formatedOriginalValue); //add value in user locale to the input
-        },
+        _onChangeTaxValueByTaxGroup(ev) {
+            let detail = ev.detail;
+            let deltaAmount = detail.oldValue-detail.newValue;
 
-        /**
-         * This method is called when the user is in edit mode and pressing 
-         * a key on his keyboard. If this key corresponds to ENTER or TAB, 
-         * the code that modifies the information is executed.
-         * 
-         * @param {event} ev 
-         */
-        _onKeydown: function (ev) {
-            switch (ev.which) {
-                // Trigger only if the user clicks on ENTER or on TAB.
-                case $.ui.keyCode.ENTER:
-                case $.ui.keyCode.TAB:
-                    // trigger blur to prevent the code being executed twice
-                    $(ev.target).blur();
+            // Search for the first tax line with the same tax group and modify its value
+            let lineId = this.record.data.line_ids.data.find(elem => elem.data.tax_group_id && elem.data.tax_group_id.data.id === detail.taxGroupId);
+
+            let debitAmount = 0;
+            let creditAmount = 0;
+            let amountCurrency = 0;
+
+            if (lineId.data.currency_id) { // If multi currency enable
+                if (this.record.data.move_type === 'in_invoice') { // For vendor bill
+                    amountCurrency = lineId.data.amount_currency - deltaAmount;
+                } else { // For refund
+                    amountCurrency = lineId.data.amount_currency + deltaAmount;
+                }
+            } else {
+                let balance = lineId.data.price_subtotal;
+                balance -= deltaAmount;
+                if (this.record.data.move_type === 'in_invoice') { // For vendor bill
+                    if (balance > 0) {
+                        debitAmount = balance;
+                    } else if (balance < 0) {
+                        creditAmount = -balance;
+                    }
+                } else { // For refund
+                    if (balance > 0) {
+                        creditAmount = balance;
+                    } else if (balance < 0) {
+                        debitAmount = -balance;
+                    }
+                }
             }
-        },
+            // Trigger the ORM
+            this.trigger('field-changed', {
+                dataPointID: this.record.id,
+                changes: { line_ids: { operation: "UPDATE", id: lineId.id, data: { amount_currency: amountCurrency, debit: debitAmount, credit: creditAmount } } },
```

Comment:
So you are telling the model that the amount currency changed but the debit/credit didn't? I would understand if you provided one or the other, but not both at once.
You are telling that debit and credit are 0 when there is a currency or am I missing something?